### PR TITLE
docs: update documentation for the caching logic and Opt-out & MaxImpression handling (SDKCF-3731)

### PR DESCRIPTION
**Pull request by imaginaris**
# Description
Includes openapi file update with new getConfig request and 429 error handling

## Links
SDKCF-3731

# Checklist
- [X] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [ ] I wrote/updated tests for new/changed code
- [X] I removed all sensitive data **before every commit**, including API endpoints and keys
- [ ] I ran `fastlane ci` without errors

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -6,34 +6,69 @@ info:
   license:
     name: MIT
     url: https://opensource.org/licenses/MIT
-  version: 3.0.0
+  version: 4.0.0
```
Comment (donnie-jp) [inline openapi.yaml]: 4.0.0 -> 4.0.0-pp

File: README.md [hunk 2]
```diff
@@ -207,6 +207,23 @@ func application(_ application: UIApplication, didFinishLaunchingWithOptions lau
 * Open `RInAppMessaging.xcworkspace` in Xcode then build/run
 * To run the tests press key shortcut command-U
 
+# **SDK Logic**
+
+## User cache
+
+Each user has a separate cache container that is persisted in UserDefaults. Each combination of rakutenId and userId is treated as a different user including a special - anonymous user - that represents non logged-in user (rakutenId and userId are null or empty).
+The key to access cache container is a SHA256 hash created from user identifiers and salt (which is a MD5 hash of user identifiers).
```
Comment (donnie-jp) [inline README.md]: I don't think an SDK consumer needs to know about our hashing implementation. Can you put that sentence on an internal confluence page please.

The rest of this section is fine 👍

**Review by donnie-jp — CHANGES_REQUESTED** [review]
See inline

File: README.md (hunk 2 above)
Comment (imaginaris) [reply README.md]: Ok. done

File: openapi.yaml (hunk 1 above)
Comment (imaginaris) [reply openapi.yaml]: Fixed (I'm leaving version number in .jazzy.yml. I think I'll add a lane in Fastfile for docs generation that will update version number in both files)

**Review by donnie-jp — APPROVED** [review]

**Pull request merged**